File: Java/Agenda/app/src/main/java/io/github/henriquemcc/agenda/java/ui/adapter/ListaAlunosAdapter.java
package io.github.henriquemcc.agenda.java.ui.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import io.github.henriquemcc.agenda.java.R;
import io.github.henriquemcc.agenda.java.model.Aluno;

public class ListaAlunosAdapter extends BaseAdapter
{

	private final List<Aluno> alunos = new ArrayList<>();
	private Context context;

	public ListaAlunosAdapter(Context context)
	{
		this.context = context;
	}

	@Override
	public int getCount()
	{
		return alunos.size();
	}

	@Override
	public Aluno getItem(int posicao)
	{
		return alunos.get(posicao);
	}

	@Override
	public long getItemId(int posicao)
	{
		return alunos.get(posicao).getId();
	}

	@Override
	public View getView(int posicao, View view, ViewGroup viewGroup)
	{
		final View viewCriada = LayoutInflater.from(context).inflate(R.layout.item_aluno, viewGroup, false);
		final Aluno alunoDevolvido = alunos.get(posicao);
		final TextView nome = viewCriada.findViewById(R.id.item_aluno_nome);
		nome.setText(alunoDevolvido.getNome());
		final TextView telefone = viewCriada.findViewById(R.id.item_aluno_telefone);
		telefone.setText(alunoDevolvido.getTelefone());
		return viewCriada;
	}

	public void clear()
	{
		alunos.clear();
	}

	public void addAll(List<Aluno> alunos)
	{
		this.alunos.addAll(alunos);
	}

	public void remove(Aluno aluno)
	{
		alunos.remove(aluno);
	}
}

File: Kotlin/Agenda/app/src/main/java/io/github/henriquemcc/agenda/kotlin/ui/adapter/ListaAlunosAdapter.kt
package io.github.henriquemcc.agenda.kotlin.ui.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import io.github.henriquemcc.agenda.kotlin.R
import io.github.henriquemcc.agenda.kotlin.model.Aluno

class ListaAlunosAdapter(private val context: Context):  BaseAdapter(){

    private val alunos = ArrayList<Aluno>()

    override fun getCount(): Int {
        return alunos.size
    }

    override fun getItem(posicao: Int): Aluno {
        return alunos[posicao]
    }

    override fun getItemId(posicao: Int): Long {
        return alunos[posicao].id.toLong()
    }

    override fun getView(posicao: Int, view: View?, viewGroup: ViewGroup?): View {
        val viewCriada = criaView(viewGroup)
        val alunoDevolvido = alunos[posicao]
        vincula(viewCriada, alunoDevolvido)
        return viewCriada
    }

    private fun criaView(viewGroup: ViewGroup?) =
        LayoutInflater.from(context).inflate(R.layout.item_aluno, viewGroup, false)

    private fun vincula(view: View, aluno: Aluno) {
        val nome = view.findViewById<TextView>(R.id.item_aluno_nome)
        nome.text = aluno.nome
        val telefone = view.findViewById<TextView>(R.id.item_aluno_telefone)
        telefone.text = aluno.telefone
    }

    fun atualiza(alunos: List<Aluno>)
    {
        this.alunos.clear()
        this.alunos.addAll(alunos)
        notifyDataSetChanged()
    }

    fun remove(aluno: Aluno)
    {
        alunos.remove(aluno)
        notifyDataSetChanged()
    }

}